langserver/workspace_refs.go

```diff
@@ -0,0 +1,173 @@
+package langserver
+
+import (
+	"context"
+	"fmt"
+	"go/build"
+	"go/token"
+	"log"
+	"sort"
+	"strings"
+	"sync"
+
+	"golang.org/x/tools/go/buildutil"
+
+	"github.com/neelance/parallel"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/sourcegraph/jsonrpc2"
+	"github.com/sourcegraph/sourcegraph-go/langserver/internal/refs"
+	"github.com/sourcegraph/sourcegraph-go/pkg/lspext"
+)
+
+func (h *LangHandler) handleWorkspaceReference(ctx context.Context, conn JSONRPC2Conn, req *jsonrpc2.Request, params lspext.WorkspaceReferenceParams) ([]lspext.ReferenceInformation, error) {
+	rootPath := h.FilePath(h.init.RootPath)
+	bctx := h.OverlayBuildContext(ctx, h.defaultBuildContext(), !h.init.NoOSFileSystemAccess)
+
```

Comment:
There is some code here duplicated with `symbols.go`. Not sure, but is it worth factoring stuff out?